extensions/cornerstone-dicom-rt/src/loadRTStruct.js

```diff
@@ -55,42 +56,54 @@ async function checkAndLoadContourData(instance, datasource) {
 
   // Modify contourData and replace it in its corresponding ROIContourSequence's Contour's contourData
   let index = 0;
-  instance.ROIContourSequence.forEach((ROIContour, roiIndex) => {
-    ROIContour.ContourSequence.forEach((Contour, contourIndex) => {
-      const promise = resolvedPromises[index++];
-
-      if (promise.status === 'fulfilled') {
-        const uint8Array = new Uint8Array(promise.value);
-        const textDecoder = new TextDecoder();
-        const dataUint8Array = textDecoder.decode(uint8Array);
-        if (
-          typeof dataUint8Array === 'string' &&
-          dataUint8Array.includes('\\')
-        ) {
-          const numSlashes = (dataUint8Array.match(/\\/g) || []).length;
-          let startIndex = 0;
-          let endIndex = dataUint8Array.indexOf('\\', startIndex);
-          let numbersParsed = 0;
-          const ContourData = [];
-
-          while (numbersParsed !== numSlashes + 1) {
-            const str = dataUint8Array.substring(startIndex, endIndex);
-            let value = parseFloat(str);
-
-            ContourData.push(value);
-            startIndex = endIndex + 1;
-            endIndex = dataUint8Array.indexOf('\\', startIndex);
-            endIndex === -1 ? (endIndex = dataUint8Array.length) : endIndex;
-            numbersParsed++;
+  instance.ROIContourSequence.forEach(ROIContour => {
+    try {
+      ROIContour.ContourSequence.forEach((Contour, contourIndex) => {
+        const promise = resolvedPromises[index++];
+        if (promise.status === 'fulfilled') {
+          if (
+            Array.isArray(promise.value) &&
+            promise.value.every(Number.isFinite)
+          ) {
+            // If promise.value is already an array of numbers, use it directly
+            Contour.ContourData = promise.value;
+          } else {
+            // Otherwise, proceed with existing logic
+            const uint8Array = new Uint8Array(promise.value);
+            const textDecoder = new TextDecoder();
+            const dataUint8Array = textDecoder.decode(uint8Array);
+            if (
+              typeof dataUint8Array === 'string' &&
+              dataUint8Array.includes('\\')
+            ) {
+              const numSlashes = (dataUint8Array.match(/\\/g) || []).length;
```

Comment:
Could you write a custom function that counts them - doing this as a regular expression works, but the efficiency isn't very good, and some of these arrays can be huge.

Reply:
Thinking about it, just try dataUint8Array.split(/\\/).map(parseFloat)  That uses a regular expression once, but then you get the entire array of strings as a result - and if it needs to be more efficient in the future, it is easy to implement an efficient version of it.